[PCI] Documentation (#245)
AdobeDocs/commerce-php: src/pages/development/security/content-security-policies.md

</csp_whitelist>
```

### Using CSP nonce provider to allow inline scripts

Adobe Commerce and Magento Open Source 2.4.7 and later include a Content Security Policy (CSP) nonce provider to facilitate the generation of unique nonce strings for each request.  These nonce strings are then attached to the CSP header.

> You can use the `generateNonce` function in `Magento\Csp\Helper\CspNonceProvider` to obtain a nonce string.

Reviewer: ```suggestion

Use the `generateNonce` function in `Magento\Csp\Helper\CspNonceProvider` to obtain a nonce string.
```

</csp_whitelist>
```

### Using CSP nonce provider to allow inline script

Beginning version 2.4.7, Magento Open Source and Adobe Commerce supports Content Security Policy (CSP) nonce. 
This enhancement introduces a CSP nonce provider, which facilitates the generation of unique nonce strings for each request. 
These nonce strings are then attached to the CSP header.

> You can utilize the `generateNonce` function in  `Magento\Csp\Helper\CspNonceProvider` in to obtain a nonce string like:

Reviewer 2: ```suggestion
You can utilize the `generateNonce` function in  `Magento\Csp\Helper\CspNonceProvider` to obtain a nonce string like:
```

</csp_whitelist>
```

> ### Using CSP nonce provider to allow inline script

Reviewer: ```suggestion
### Using CSP nonce provider to allow inline scripts
```

### Report-Uri configuration

Regardless of `restrict` or `report-only` mode, CSP violations may be reported to an endpoint for collection.

1. On the _Admin_ sidebar, go to **Stores** > _Settings_ > **Configuration**.

   If you have a multi-site installation, set the Store View control in the upper-left corner to the website where the configuration applies.

> 2. In the left panel under _Security_, choose **Content Security Policy**

Reviewer: ```suggestion
2. In the left panel under _Security_, choose **Content Security Policy**.
```

![](../../_images/security/csp-report-uri.png)

> The create order page in the Admin and the checkout page in the storefront can be configured to have their own URI. However, if these fields are left blank then the default URI for the admin and storefront areas is used for reporting.

Reviewer: ```suggestion
The Create Order page in the Admin and the Checkout page in the storefront can be configured to have their own URI. However, if these fields are left blank, the default URI for the admin and storefront areas is used for reporting.
```

### Page specific Content-Security-Policies

Adobe Commerce and Magento Open Source can send unique policies for a specific page.

> For example, to configure policies for the checkout page in the storefront, add the following to the `config.xml` file for your custom module:

Reviewer: ```suggestion
For example, to configure policies for the Checkout page in the storefront, add the following to the `config.xml` file for your custom module:
```

### Page specific Content-Security-Policies

Adobe Commerce and Magento Open Source can send unique policies for a specific page.

> For example, to configure policies for the checkout page in the storefront, add the following in your custom module's `config.xml`

Reviewer: ```suggestion
For example, to configure policies for the checkout page in the storefront, add the following to the `config.xml` file for your custom module:
```

You do not need to define other policy options in contexts like _unsafe-inline_. The same
policy options read from config will be merged later.

### Troubleshooting

Users might sometimes see browser errors due to certain scripts being blocked because of CSP:

`Refused to execute inline script because it violates the following Content Security Policy directive: "script-src`

> To fix this issue, you must [whitelist](https://developer.adobe.com/commerce/php/development/security/content-security-policies/#whitelist-an-inline-script-or-style) the blocked scripts using the `SecureHtmlRenderer` class or use the [CSPNonceProvider](https://developer.adobe.com/development/security/content-security-policies/#using-csp-nonce-provider-to-allow-inline-script) class to allow scripts to be executed.

Reviewer: ```suggestion
To fix this issue, [whitelist](https://developer.adobe.com/commerce/php/development/security/content-security-policies/#whitelist-an-inline-script-or-style) the blocked scripts using the `SecureHtmlRenderer` class or use the [CSPNonceProvider](https://developer.adobe.com/development/security/content-security-policies/#using-csp-nonce-provider-to-allow-inline-script) class to allow scripts to be executed.
```
